File: tdd-kit/tdd-kit-cpp/sources/fizzbuzz/fizzbuzz.cpp
#include <stdio.h>

#include "fizzbuzz.h"

string fizzbuzz(const int number) {
	if (number % 15 == 0) {
		return "FizzBuzz";
	} else if (number % 3 == 0) {
		return "Fizz";
	} else if (number % 5 == 0) {
		return "Buzz";
	}
	return to_string(number);
}

File: tdd-kit/tdd-kit-csharp/TddKitCsharp/Fizzbuzz.cs
using System;
using System.Collections.Generic;

namespace TddKitCsharp
{
    public class Fizzbuzz
    {
        public IList<string> Elements()
        {
            IList<string> elements = new List<string>();
            for (var i = 1; i <= 100; i++)
            {
                elements.Add(ValueOf(i));
            }
            return elements;
        }

        private static string ValueOf(int number)
        {
            if (number%15 == 0)
            {
                return "fizzbuzz";
            }
            if (number%3 == 0)
            {
                return "fizz";
            }
            if (number%5 == 0)
            {
                return "buzz";
            }
            return Convert.ToString(number);
        }
    }
}